utils/python/CIME/SystemTests/erio.py

```diff
@@ -0,0 +1,115 @@
+"""
+ERIO tests restart with different PIO methods
+
+This class inherits from SystemTestsCommon
+"""
+from CIME.XML.standard_module_setup import *
+from CIME.SystemTests.system_tests_common import SystemTestsCommon
+import shutil
+
+logger = logging.getLogger(__name__)
+
+class ERIO(SystemTestsCommon):
+
+    def __init__(self, case):
+        """
+        initialize an object interface to file env_test.xml in the case directory
+        """
+        SystemTestsCommon.__init__(self, case, expected=["TEST"])
+
+        self._pio_types = self._case.get_env("run").get_valid_values("PIO_TYPENAME")
+
+    def build_phase(self, sharedlib_only=False, model_only=False):
+        """
+        Build one executable per PIO method
+        """
+        exeroot = self._case.get_value("EXEROOT")
+        cime_model = self._case.get_value("MODEL")
+
+        for pio_type in self._pio_types:
+            if pio_type == "default":
+                continue
+            else:
+                self._case.set_value("PIO_TYPENAME", pio_type)
+                self._case.flush()
+
+                self.clean_build()
+                self.build_indv(sharedlib_only=sharedlib_only, model_only=model_only)
```

Comment:
Not sure if PIO if the case.set_value above is having any impact on the PIO build here.